src/__tests__/waitFor.test.tsx

```diff
@@ -204,6 +204,43 @@ test.each([false, true])(
   }
 );
 
+const blockThread = (timeToBlockThread: number, legacyFakeTimers: boolean) => {
+  jest.useRealTimers();
+  let end = Date.now() + timeToBlockThread;
+
+  while (Date.now() < end) {
+    // do nothing
+  }
+
+  jest.useFakeTimers({ legacyFakeTimers });
+};
+
+test.each([true, false])(
+  'it should not depend on real time when using fake timers (legacyFakeTimers = %s)',
+  async (legacyFakeTimers) => {
+    jest.useFakeTimers({ legacyFakeTimers });
+
+    const mockErrorFn = jest.fn(() => {
+      // Wait 5 seconds so that check time is longer than interval
+      blockThread(5, legacyFakeTimers);
```

Comment:
I would wait a little bit longer to make the test more robust using the exact value might cause some race conditions
```suggestion
      blockThread(8, legacyFakeTimers);
```

Reply:
Actually I'm not using the exact value, the aim is that we block the thread for a time longer than the interval, but I'll make it a bit more explicit